streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -324,6 +329,22 @@ void close() {
         transactionInitialized = false;
     }
 
+    /**
+     * Disables producer reset to prevent producer recreation during shutdown.
+     * <p>
+     * When disabled, subsequent calls to reInitializeProducer() will not recreate
+     * the producer instance, avoiding resource leak.
+     * <p>
+     * <strong>
+     *     This method should only be invoked when the {@link org.apache.kafka.streams.processor.internals.ActiveTaskCreator}
```

Comment:
nit: I'd say when the "StreamThread" is shutting down, rather than the ActiveTaskCreator

Effectively it's the same thing at the moment but you never know what refactoring may bring, and the important thing here is that the StreamThread itself is shutting down

Reply:
You are right.
I will clarify it in the next commit